erdpy/proxy/core.py

```diff
@@ -140,24 +143,19 @@ def get_hyperblock(self, key) -> Any:
         response = response.get("hyperblock", {})
         return response
 
-    def send_transaction_and_wait_for_result(self, payload: Any, num_seconds_timeout: int = 100) -> Dict[str, Any]:
+    def send_transaction_and_wait_for_result(self, payload: Any, num_seconds_timeout: int = 100) -> ITransactionOnNetwork:
         url = f"{self.url}/transaction/send"
         response = do_post(url, payload)
         tx_hash = response.get("txHash")
 
         for _ in range(0, num_seconds_timeout):
-            if self.is_transaction_finalized(tx_hash):
-                return self.get_transaction(tx_hash=tx_hash, with_results=True)
-            time.sleep(1)
-        return dict()
-
-    def is_transaction_finalized(self, tx_hash):
-        last_nonce = self.get_last_block_nonce("metachain")
-        last_hyperblock = self.get_hyperblock(last_nonce)
-        finalized_transactions = last_hyperblock["transactions"]
+            time.sleep(AWAIT_TRANSACTION_PERIOD)
```

Comment:
As it's written right now, this now waits for `100 * 5 = 500` seconds instead of 100, as the argument suggests.

Reply:
:tada: as always, very very nice catch!